From a14243189a3be25e3d7dc9616f9af28255be073c Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Fri, 18 Jun 2021 19:10:55 +0000
Subject: [PATCH 14/14] XANMOD: Makefile: Turn off tree vectorization for GCC
 -O3 optimization level

Signed-off-by: Alexandre Frade <kernel@xanmod.org>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index df92892325ae..9557284557e0 100644
--- a/Makefile
+++ b/Makefile
@@ -757,6 +757,7 @@ ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
 KBUILD_CFLAGS += -O2
 else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O3
 KBUILD_CFLAGS += -O3
+KBUILD_CFLAGS += $(call cc-option, -fno-tree-vectorize)
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS += -Os
 endif
-- 
2.35.1

