From f3d51167778023041407f339c2252686445452c5 Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Wed, 4 Aug 2021 12:32:24 +0000
Subject: [PATCH 11/11] fs/ntfs3: Fix unsupported flags by clang

Signed-off-by: Alexandre Frade <kernel@xanmod.org>
---
 fs/ntfs3/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/ntfs3/Makefile b/fs/ntfs3/Makefile
index b06a06cc006f..279701b62bbe 100644
--- a/fs/ntfs3/Makefile
+++ b/fs/ntfs3/Makefile
@@ -4,7 +4,9 @@
 #
 
 # to check robot warnings
-ccflags-y += -Wunused-but-set-variable -Wold-style-declaration -Wint-to-pointer-cast
+ccflags-y += -Wint-to-pointer-cast \
+	$(call cc-option,-Wunused-but-set-variable,-Wunused-const-variable) \
+	$(call cc-option,-Wold-style-declaration,-Wout-of-line-declaration)
 
 obj-$(CONFIG_NTFS3_FS) += ntfs3.o
 
-- 
2.32.0

