From 34c89a08f5d7ed549da33c6802a85cda003af0ee Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Thu, 3 Sep 2020 20:36:13 +0000
Subject: [PATCH 12/14] XANMOD: init/Kconfig: Enable -O3 KBUILD_CFLAGS
 optimization for all architectures

Signed-off-by: Alexandre Frade <kernel@xanmod.org>
---
 init/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/init/Kconfig b/init/Kconfig
index b52bb739bfa6..5a1598cb1c6a 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1346,7 +1346,6 @@ config CC_OPTIMIZE_FOR_PERFORMANCE
 
 config CC_OPTIMIZE_FOR_PERFORMANCE_O3
 	bool "Optimize more for performance (-O3)"
-	depends on ARC
 	help
 	  Choosing this option will pass "-O3" to your compiler to optimize
 	  the kernel yet more for performance.
-- 
2.32.0

