From aba4b736ea8fcd544bdd34c954a3721f2ab1192c Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Wed, 30 Jun 2021 19:40:01 +0000
Subject: [PATCH] mm, futex2: fix shared futex2 pgoff on shmem huge page

Signed-off-by: Alexandre Frade <kernel@xanmod.org>
---
 kernel/futex2.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernel/futex2.c b/kernel/futex2.c
index 506cb9ecc143..3caed3e6894c 100644
--- a/kernel/futex2.c
+++ b/kernel/futex2.c
@@ -14,7 +14,6 @@
  */
 
 #include <linux/freezer.h>
-#include <linux/hugetlb.h>
 #include <linux/jhash.h>
 #include <linux/memblock.h>
 #include <linux/pagemap.h>
@@ -310,7 +309,7 @@ static int futex_get_shared_key(uintptr_t address, struct mm_struct *mm,
 		}
 
 		key->pointer = futex_get_inode_uuid(inode);
-		key->index = (unsigned long)basepage_index(tail);
+		key->index = (unsigned long)page_to_pgoff(tail);
 		key->offset |= FUT_OFF_INODE;
 
 		rcu_read_unlock();
-- 
2.32.0

