From 00c3cc4ac0aa590344b95094a73793acbfc07a2d Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Wed, 5 Jan 2022 19:58:06 +0000
Subject: [PATCH 01/12] XANMOD: block/mq-deadline: Set sequential requests for lower
 latency

Signed-off-by: Alexandre Frade <kernel@xanmod.org>
---
 block/mq-deadline.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/mq-deadline.c b/block/mq-deadline.c
index e4e90761eab3..8fc3aa5a54d8 100644
--- a/block/mq-deadline.c
+++ b/block/mq-deadline.c
@@ -30,7 +30,7 @@
 static const int read_expire = HZ / 2;  /* max time before a read is submitted. */
 static const int write_expire = 5 * HZ; /* ditto for writes, these limits are SOFT! */
 static const int writes_starved = 2;    /* max times reads can starve a write */
-static const int fifo_batch = 16;       /* # of sequential requests treated as one
+static const int fifo_batch = 8;       /* # of sequential requests treated as one
 				     by the above parameters. For throughput. */
 
 enum dd_data_dir {
-- 
2.33.1

