From a6c8115910440f8ffef10c45346d6eeedde55293 Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Fri, 18 Jun 2021 19:10:55 +0000
Subject: [PATCH 11/12] XANMOD: Makefile: Turn off loop vectorization for GCC
 -O3 optimization level

Signed-off-by: Alexandre Frade <kernel@xanmod.org>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index ed6e7ec60eff..346f0d25bfa6 100644
--- a/Makefile
+++ b/Makefile
@@ -755,6 +755,7 @@ ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
 KBUILD_CFLAGS += -O2
 else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O3
 KBUILD_CFLAGS += -O3
+KBUILD_CFLAGS += $(call cc-option, -fno-tree-loop-vectorize)
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS += -Os
 endif
-- 
2.33.1

