From bb5cb7d5b5fe4a742a0ca1cc90ff301cf073eea5 Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Mon, 28 Aug 2023 05:00:29 +0000
Subject: [PATCH 02/18] XANMOD: x86/build: Add more x86 code optimization flags

Signed-off-by: Alexandre Frade <kernel@xanmod.org>
---
 arch/x86/Makefile | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index d6f234e16626..51ab6ffbbe51 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -63,6 +63,12 @@ export REALMODE_CFLAGS
 # e.g.: obj-y += foo_$(BITS).o
 export BITS
 
+# More compiler optimizations
+ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
+KBUILD_CFLAGS += -O3 $(call cc-option,-fmodulo-sched -fivopts)
+KBUILD_RUSTFLAGS += -Copt-level=3
+endif
+
 #
 # Prevent GCC from generating any FP code by mistake.
 #
-- 
2.43.0

