From 76d7ea767006e1caa29de160f4e2fb9a81bff990 Mon Sep 17 00:00:00 2001
From: Hamad Al Marri <hamad@cachyos.org>
Date: Wed, 30 Jun 2021 18:03:00 +0300
Subject: [PATCH 2/2] XANMOD: cacule: remove delta since it is not used.

---
 kernel/sched/core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index e31b58b642c2..65b615b01700 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -4680,7 +4680,9 @@ static void sched_tick_remote(struct work_struct *work)
 	struct rq *rq = cpu_rq(cpu);
 	struct task_struct *curr;
 	struct rq_flags rf;
+#if !defined(CONFIG_CACULE_SCHED)
 	u64 delta;
+#endif
 	int os;
 
 	/*
-- 
2.32.0

